src/TRestRawSignal.cxx

```diff
@@ -631,21 +631,66 @@ void TRestRawSignal::GetSignalSmoothed(TRestRawSignal* smthSignal, Int_t averagi
 /// \param averagingPoints It defines the number of neightbour consecutive
 /// points used to average the signal
 ///
-std::vector<Float_t> TRestRawSignal::GetSignalSmoothed(Int_t averagingPoints) {
+/// \param option If the option is set to "EXCLUDE OUTLIERS", points that are too far away from the median baseline will be ignored to improve the smoothing result
+///
+std::vector<Float_t> TRestRawSignal::GetSignalSmoothed(Int_t averagingPoints, std::string option) {
+    
     std::vector<Float_t> result;
 
+    if (option == ""){
+        
+        averagingPoints = (averagingPoints / 2) * 2 + 1;  // make it odd >= averagingPoints
+
+        Float_t sumAvg = (Float_t)GetIntegralInRange(0, averagingPoints) / averagingPoints;
+
+        for (int i = 0; i <= averagingPoints / 2; i++) result.push_back(sumAvg);
+
+        for (int i = averagingPoints / 2 + 1; i < GetNumberOfPoints() - averagingPoints / 2; i++) {
+            sumAvg -= this->GetRawData(i - (averagingPoints / 2 + 1)) / averagingPoints;
+            sumAvg += this->GetRawData(i + averagingPoints / 2) / averagingPoints;
+            result.push_back(sumAvg);
```

Comment:
In general is better to avoid `push_back` if not required, you already know the size of the `vector` right? Then I would suggest to initialize at the beginning of the `if` `result.resize(GetNumberOfPoints())`